main.go

```diff
@@ -132,6 +148,14 @@ func main() {
 	}
 
 	server.HandlePublish = func(conn *rtmp.Conn) {
+		connectingKey := strings.ReplaceAll(conn.URL.Path, "/", "")
+
+		if connectingKey != *streamKey {
+			fmt.Println("Connection attempt made using incorrect key: ", connectingKey)
+			conn.Close()
+			return
+		}
```

Comment:
```suggestion
		if !strings.Contains(conn.URL.Path, fmt.Sprintf("/%s", *streamKey)) {
			fmt.Println("Connection attempt made using incorrect key.  Path Provided: ", conn.URL.Path)
			conn.Close()
			return
		}
```

Hows something like this?  Maybe a bit simpler and friendlier to key being located in url?

Reply:
Maybe even simpler:

```go
if !strings.HasSuffix(conn.URL.Path, *streamKey) {
...
```

Reply:
ah.. even easier.  I like it